From f8d051f3dea0302b9305454b9a9c3f279ef48fc6 Mon Sep 17 00:00:00 2001
From: Topi Reinio <topi.reinio@qt.io>
Date: Tue, 16 Sep 2025 11:08:48 +0000
Subject: [PATCH 237/239] Doc: Qml Compiler: Fix internal documentation issues

Mark QQmlSA::PropertyPrivate \internal to fix QDoc warning:
  src/qmlcompiler/qqmlsa.cpp:727: (qdoc) warning: No output generated
    for function 'QQmlSA::PropertyPrivate::isReadonly()' because
    'PropertyPrivate' is undocumented

Move the \class topic for QQmlJSMetaPropertyBinding from a header to
the .cpp file. Fixes:

  src/qmlcompiler/qqmljsmetatypes.cpp:53: (qdoc) warning: No output
    generated for function
    'QQmlJSMetaPropertyBinding::translationDataValue()' because
    'QQmlJSMetaPropertyBinding' is undocumented

Mark QQmlJSImporter \internal, fixes

  src/qmlcompiler/qqmljsimporter.cpp:719: (qdoc) warning: No output
  generated for function 'QQmlJSImporter::importQmldirs()' because
  'QQmlJSImporter' is undocumented.

Change-Id: I18f66b096b6f30941f55f5cfdae4800cafc0a51a
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
Reviewed-by: Paul Wicking <paul.wicking@qt.io>
(cherry picked from commit 27f66343163b6d565c93f3969459d350da812d13)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlcompiler/qqmljsimporter.cpp  |  4 ++++
 src/qmlcompiler/qqmljsmetatypes.cpp | 10 +++++++++-
 src/qmlcompiler/qqmljsmetatypes_p.h |  9 ---------
 src/qmlcompiler/qqmlsa.cpp          |  4 ++++
 4 files changed, 17 insertions(+), 10 deletions(-)

diff --git a/src/qmlcompiler/qqmljsimporter.cpp b/src/qmlcompiler/qqmljsimporter.cpp
index 39286e9ba0..290738ed8f 100644
--- a/src/qmlcompiler/qqmljsimporter.cpp
+++ b/src/qmlcompiler/qqmljsimporter.cpp
@@ -164,6 +164,10 @@ static QStringList aliases(const QQmlJSScope::ConstPtr &scope)
         : scope->aliases();
 }
 
+/*!
+    \class QQmlJSImporter
+    \internal
+*/
 QQmlJSImporter::QQmlJSImporter(const QStringList &importPaths, QQmlJSResourceFileMapper *mapper,
                                QQmlJSImporterFlags flags)
     : m_importPaths(importPaths),
diff --git a/src/qmlcompiler/qqmljsmetatypes.cpp b/src/qmlcompiler/qqmljsmetatypes.cpp
index 8233f21225..843527ea96 100644
--- a/src/qmlcompiler/qqmljsmetatypes.cpp
+++ b/src/qmlcompiler/qqmljsmetatypes.cpp
@@ -10,7 +10,16 @@
 QT_BEGIN_NAMESPACE
 
 /*!
+    \class QQmlJSMetaPropertyBinding
+
     \internal
+
+    Represents a single QML binding of a specific type. Typically, when you
+    create a new binding, you know all the details of it already, so you should
+    just set all the data at once.
+*/
+
+/*!
     A binding is valid when it has both a target (m_propertyName is set)
     and some content set (m_bindingType != Invalid).
 */
@@ -74,7 +83,6 @@ QQmlTranslation QQmlJSMetaPropertyBinding::translationDataValue(QString qmlFileN
 }
 
 /*!
-    \internal
     Uses \a resolver to return the correct type for the stored literal
     and a null scope pointer if the binding does not contain a literal
  */
diff --git a/src/qmlcompiler/qqmljsmetatypes_p.h b/src/qmlcompiler/qqmljsmetatypes_p.h
index 6df8b22ed0..d5b2704f44 100644
--- a/src/qmlcompiler/qqmljsmetatypes_p.h
+++ b/src/qmlcompiler/qqmljsmetatypes_p.h
@@ -512,15 +512,6 @@ public:
     }
 };
 
-/*!
-    \class QQmlJSMetaPropertyBinding
-
-    \internal
-
-    Represents a single QML binding of a specific type. Typically, when you
-    create a new binding, you know all the details of it already, so you should
-    just set all the data at once.
-*/
 class Q_QMLCOMPILER_EXPORT QQmlJSMetaPropertyBinding
 {
     using BindingType = QQmlSA::BindingType;
diff --git a/src/qmlcompiler/qqmlsa.cpp b/src/qmlcompiler/qqmlsa.cpp
index 751689c840..5797701270 100644
--- a/src/qmlcompiler/qqmlsa.cpp
+++ b/src/qmlcompiler/qqmlsa.cpp
@@ -688,6 +688,10 @@ QQmlJSMetaMethod MethodPrivate::method(const QQmlSA::Method &method)
     return method.d_func()->m_method;
 }
 
+/*!
+    \class QQmlSA::PropertyPrivate
+    \internal
+*/
 PropertyPrivate::PropertyPrivate(Property *iface) : q_ptr{ iface } { }
 
 PropertyPrivate::PropertyPrivate(Property *iface, const PropertyPrivate &other)
-- 
2.51.2

